hack/get-server.sh

```diff
@@ -1,55 +1,106 @@
+esac
+
 # Check machine architecture
 ARCH=$(uname -m)
 # Check operating system
 OS=$(uname -s)
 
-if [ "$OS" == "" ]; then
-  OS=$(ver)
-  ARCH=$(echo %PROCESSOR_ARCHITECTURE%)
-fi
+case $OS in
+  "Darwin")
+    FILENAME="darwin"
+    ;;
+  "Linux")
+    FILENAME="linux"
+    ;;
+  *)
+    err "Unsupported operating system: $OS"
+    ;;
+esac
 
-if [ "$OS" == "Darwin" ]; then
-  FILENAME="darwin"
-elif [ "$OS" == "Linux" ]; then
-  FILENAME="linux"
-elif [[ $OS == *"Windows"* ]]; then
-  FILENAME="windows"
-else
-  echo "Unsupported operating system: $OS"
-  exit 1
-fi
+case $ARCH in
+  "arm64" | "ARM64")
+    FILENAME="$FILENAME-arm64"
+    ;;
+  "x86_64" | "AMD64")
+    FILENAME="$FILENAME-amd64"
+    ;;
+  "aarch64")
+    FILENAME="$FILENAME-arm64"
+    ;;
+  *)
+    err "Unsupported architecture: $ARCH"
+    ;;
+esac
 
-if [ "$ARCH" == "arm64" ]; then
-  FILENAME=$(echo "$FILENAME-arm64")
-elif [ "$ARCH" == "ARM64" ]; then
-  FILENAME=$(echo "$FILENAME-arm64")
-elif [ "$ARCH" == "x86_64" ]; then
-  FILENAME=$(echo "$FILENAME-amd64")
-elif [ "$ARCH" == "AMD64" ]; then
-  FILENAME=$(echo "$FILENAME-amd64")
-elif [ "$ARCH" == "aarch64" ]; then
-  FILENAME=$(echo "$FILENAME-arm64")
-else
-  echo "Unsupported architecture: $ARCH"
-  exit 1
-fi
+DOWNLOAD_URL="$BASE_URL/$VERSION/daytona-$FILENAME"
 
-BASE_URL="https://download.daytona.io/daytona"
-if [ -n "$DAYTONA_SERVER_DOWNLOAD_URL" ]; then
-  BASE_URL=$DAYTONA_SERVER_DOWNLOAD_URL
-fi
+echo "Downloading server from $DOWNLOAD_URL"
 
-DOWNLOAD_URL=$(echo "$BASE_URL/$VERSION/daytona-$FILENAME")
+# Create a temporary file to download the server binary. Just in case the author... user
+# has say a directory named "daytona" in $HOME... the current directory.
+temp_file="daytona-$RANDOM"
 
-echo "Downloading server from $DOWNLOAD_URL"
+# Ensure the temporary file is deleted on exit
+trap 'rm -f "$temp_file"' EXIT
+
+# curl does not fail with a non-zero status code when the HTTP status
+# code is not successful (like 404 or 500). You can add -f flag to curl
+# command to fail silently on server errors.
+#
+# Flags:
+# -f, --fail: Fail silently on server errors.
+# -s, --silent: Silent mode. Don't show progress meter or error messages.
+# -S, --show-error: When used with -s, show error even if silent mode is enabled.
+# -L, --location: Follow redirects.
+# -o, --output <file>: Write output to <file> instead of stdout.
+curl -fsSL "$DOWNLOAD_URL" -o "$temp_file"
+if [ $? -ne 0 ]; then
+  err "Daytona server download failed"
+fi
+chmod +x "$temp_file"
+
+echo "Installing server to $DESTINATION"
+# Check if sudo is required
+if [ "$SUDO_REQUIRED" -eq 1 ] && [ "$EUID" -ne 0 ]; then
+  err "Cannot write to /opt/bin or /usr/local/bin as non root. Please re-run with sudo."
+fi
 
-sudo curl $DOWNLOAD_URL -Lo daytona
-sudo mv daytona /usr/local/bin/daytona
-sudo chmod +x /usr/local/bin/daytona
\ No newline at end of file
+mv "$temp_file" "$DESTINATION"
```

Comment:
I think this will move daytona binary with the name `daytona-$RANDOM` into specified PATH. So this would fix it
```
mv "$temp_file" "$DESTINATION/daytona"
```

Reply:
@zzorica Amazing catch. Fixed.